### ISSUE TITLE
Question: What is meant by "partially filled block" and "arbitrary flush"?

### ISSUE THREAD

**Issue by ecm-pushbx** [lz4/lz4#623]
In https://github.com/lz4/lz4/blob/e5a1911ec298a03242b2c8f954f3b511d5a743f6/doc/lz4_Frame_format.md in the section Data Blocks, subsection Data, it reads:

> Note that data block is not necessarily full : an arbitrary “flush” may happen anytime. Any block can be “partially filled”.

What is meant by that? When implementing my decoder (strictly from the documented Block and Frame formats, not porting an existing decoder) I did not figure out what a "partially filled" block would look like. After each sequence (literal + match) and after each literals string, I check whether I'm at the end of that block (as determined by the Block Size given at the block's beginning).

So, how is a block supposed to look if it's "partially filled"? Or is this an obsolete artefact in the spec?

**Comment by Cyan4973** [comment]
It's probably poor wording.

The word "full" refers to the maximum size that a block can have in the current frame.
This maximum was defined previously in the [frame header](https://github.com/lz4/lz4/blob/dev/doc/lz4_Frame_format.md#frame-descriptor).

"partially filled" simply means that no block is guaranteed to contain the maximum nb of bytes as described in the frame header. Hence the decompressed size of a block is not known in advance.

I can rephrase this paragraph if it helps.

**Comment by ecm-pushbx** [comment]
Thanks for the explanation! I'd reword it as follows:

> Note that a data block is not necessarily full, that is, it may contain less data than the maximum block size (as specified in the BD Byte).

**Comment by Cyan4973** [comment]
Updated frame format specification, in an attempt to clarify existence of non-full blocks.
